module/filter/VecToScalarVtkm/VecToScalarVtkm.h

```diff
@@ -0,0 +1,33 @@
+#ifndef VISTLE_VECTOSCALARVTKM_VECTOSCALARVTKM_H
+#define VISTLE_VECTOSCALARVTKM_VECTOSCALARVTKM_H
+
+#include <vistle/vtkm/vtkm_module.h>
+
+
+class VecToScalarVtkm: public vistle::VtkmModule {
```

Comment:
I don't think that the VtkmModule base class is a good match for this task: In the case where a component needs to be extracted, the module does not have to perform any work, but it just needs to copy over the reference to the single component of the data. And this should not require an implementation relying on Viskores. So, keeping what VecToScalar does for this case should be fine.

Only if the magnitude of the vector has to be computed, Viskores should get involved. And for that it should be sufficient to provide an alternative implementation of calculateAbsolute. This implementation should make use of the methods provided by vistle_vtkm library to take care of the conversion between Viskores and Vistle data.